objects/src/assets/mod.rs

```diff
@@ -139,6 +159,11 @@ impl FungibleAsset {
         self.amount
     }
 
+    /// Returns true if this and the other assets were issued from the same faucet.
+    pub fn is_same(&self, other: &Self) -> bool {
```

Comment:
nit: `const fn`

Also; should we rename it to something unmistakable? Like `has_same_faucet_than`

Reply:
I don't think we can use `const fn` here because we use `==` for comparison of IDs, and that is not `const fn`.

Agree that a better name can be used here: I renamed it to `from_same_faucet()`.

Reply:
Maybe prefix with `is_`? It might be a quite long name, but prefixing with `from` suggests we will create a new instance of `Self` when we are computing something over an existing one.

Reply:
For reference: https://github.com/rust-lang/rfcs/blob/master/text/0430-finalizing-naming-conventions.md#general-naming-conventions

Reply:
Agreed! Renamed into `is_from_same_faucet`.